src/Common/ProgressBar.cpp

```diff
@@ -0,0 +1,161 @@
+#include "ProgressBar.h"
+#include <IO/WriteBufferFromFileDescriptor.h>
+#include <Common/TerminalSize.h>
+#include <Common/UnicodeBar.h>
+#include <Databases/DatabaseMemory.h>
+
+namespace DB
+{
+bool ProgressBar::getNeedRenderProgress() const
+{
+    return need_render_progress;
+}
+
+bool ProgressBar::getShowProgressBar() const
+{
+    return show_progress_bar;
+}
+
+size_t ProgressBar::getWrittenProgressChars() const
+{
+    return written_progress_chars;
+}
+
+bool ProgressBar::getWrittenFirstBlock() const
+{
+    return written_first_block;
+}
+
+bool ProgressBar::getClearProgress() const
+{
+    return clear_progress;
+}
+
+void ProgressBar::setNeedRenderProgress(bool needRenderProgress)
+{
+    need_render_progress = needRenderProgress;
+}
+
+void ProgressBar::setShowProgressBar(bool showProgressBar)
+{
+    show_progress_bar = showProgressBar;
+}
+
+void ProgressBar::setWrittenProgressChars(size_t writtenProgressChars)
+{
+    written_progress_chars = writtenProgressChars;
+}
+
+void ProgressBar::setWrittenFirstBlock(bool writtenFirstBlock)
+{
+    written_first_block = writtenFirstBlock;
+}
+
+void ProgressBar::setClearProgress(bool clearProgress)
+{
+    clear_progress = clearProgress;
+}
+
+bool ProgressBar::updateProgress(Progress &progress, const Progress &value)
+{
+    return progress.incrementPiecewiseAtomically(value);
+}
```

Comment:
This method looks strange. Why not just put `progress` inside `ProgressBar` class?

Reply:
But `watch` and `progress` are used in other places in `Client.cpp`. If I put them inside the class, I am gonna need to use getter functions in `Cleint.cpp`.  Is it okay? Or is there another way?